navapbc/wic-mt-demo-project-eligibility-screener
# Refactor (8/8): Cleanup & e2e tests

## Ticket

Came out of refactor work tied to:
- https://wicmtdp.atlassian.net/browse/WMDP-236

## Changes
> What was added, updated, or removed in this PR.

- Remove the no longer used state context & its path alias
- Add end-to-end tests using [playwright](https://playwright.dev)

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

### Stacked PR Overview

This is part of a series of stacked PRs that break down an overly-large chunk of work that originated in refactoring work. The original branch and draft PR is https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/pull/136  

These stacked PRs are not semantic PRs (i.e. they do not contain work organized by feature) as that would have been very difficult to recreate out of the overly-large chunk of work. Instead, these PRs are broken down to accomplish the following goals:

1. Make the PRs small enough to be reviewable
2. Group similar work together to make it easy to review
3. Commit dependencies first (i.e. low number PRs have dependencies and packages, high number PRs have complex components and pages)
4. Make each PR pass CI automated testing
5. Make each file that was changed only included in a single PR to reduce possibility for merge/rebase conflicts

### Recommended PR Review Strategy

The commits in each PR have been structured to group similar work together, so I recommend using Github's PR "commits" tab to review each PR commit-by-commit. 

### Stacked PR 8/8 

This PR (8 of 8) is the cleanup PR. It removes the state context that was refactored out in #146. It adds end-to-end tests to handle the gated page access in #142. 

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

These changes can be tested using...

1. Install the packages with `yarn install`
5. Run the automated unit tests with `yarn test`
6. Build the docker image and start the container with `docker-compose up -d --build` (playwright assumes the app is running at localhost:3000)
7. Install playwright with `npx playwright install`
8. Run the e2e tests with `yarn test-e2e`
5. Shut down the container with `docker-compose down`
